commons/src/main/java/io/hyscale/commons/exception/HyscaleException.java

```diff
@@ -19,45 +19,45 @@
 
 public class HyscaleException extends Exception {
 
-    private HyscaleErrorCode hyscaleErrorCode;
+    private HyscaleError hyscaleError;
     private String[] args;
     private Integer code = ToolConstants.HYSCALE_ERROR_CODE;
```

Comment:
Can we do away with code variable? In getcode() we can keep the check and return default  value if original value is not available.

Reply:
There are 2 constructors which are taking custom code as input parameter. So I think it can't be achieved if we remove that variable.

public HyscaleException(HyscaleError hyscaleError, Integer code) {
        this.hyscaleError = hyscaleError;
        this.code = code;
    }

    public HyscaleException(HyscaleError hyscaleError, Integer code, String... args) {
        this.hyscaleError = hyscaleError;
        this.code = code;
        this.args = args;
    }

Reply:
You can remove those constructors, we only kept them as a way around to have exit code in HyscaleException. You can also verify that they are not used anywhere.

Reply:
HyscaleException.HyscaleException(HyscaleError, Integer)  (io.hyscale.commons.exception)   is being used in 
SpecSchemaValidator.validate(File)(3 usages)  (io.hyscale.controller.validator)
InputSpecValidator.validateFile(File)  (io.hyscale.controller.validator)
HyscaleException.HyscaleException(HyscaleError, Integer, String...)  (io.hyscale.commons.exception) is being used in 
ValidatorMessageHandler.handleErrMsg(String, boolean, boolean)  (io.hyscale.controller.util)
InputSpecValidator.validateFile(File)(4 usages)  (io.hyscale.controller.validator)


Does it required to remove these constructors and refactor the references of the above ..

Reply:
For now leave the constructor as is. We can think on this later.